bot/commands.ts

```diff
@@ -129,29 +133,87 @@ const commands: Record<string, Command> = {
 			contexts: [0, 2],
 		},
 		execute: async (interaction) => {
-			if (interaction?.guildId) {
-				const guild = interaction.guild?.id
-				const user = interaction.user.id
-				const xp = await makeGETRequest(guild as string, user)
-
-				if (!xp) {
-					await interaction.reply({
-						ephemeral: true,
-						content: "No XP data available."
-					});
-					return;
-				}
-
-				const progress = xp.user_progress_next_level;
-				const progressBar = createProgressBar(progress);
+			const guild = interaction.guild?.id
+			const user = interaction.user.id
+			const xp = await makeGETRequest(guild as string, user)
 
+			if (!xp) {
 				await interaction.reply({
-					embeds: [
+					ephemeral: true,
+					content: "No XP data available."
+				});
+				return;
+			}
+			
+			const card = new RankCardBuilder()
+  			.setDisplayName((interaction.member as GuildMember).displayName)
+        .setAvatar(interaction.user.displayAvatarURL()) // user avatar
+        .setCurrentXP(300) // current xp
+        .setRequiredXP(600) // required xp
+        .setLevel(2) // user level
+        .setRank(5) // user rank
+        .setOverlay(90) // overlay percentage. Overlay is a semi-transparent layer on top of the background
+        .setBackground("#23272a")
```

Comment:
Might require a custom library but use the banner for nitro users. Or easier, use their theme for the background?

Reply:
I'm not sure if d.js has the theme color, but you are right with using banners for the background instead, we should probably consider it for the future